internal/application/generator.go

```diff
@@ -170,8 +170,10 @@ func (g *Generator) apply(app *redskyappsv1alpha1.Application) error {
 		app.Scenarios = append(app.Scenarios, *s)
 	}
 
-	for _, o := range g.Objectives {
-		app.Objectives = append(app.Objectives, redskyappsv1alpha1.Objective{Name: o})
+	if o, err := g.readObjective(); err != nil {
+		return err
+	} else if o != nil {
+		app.Objectives = append(app.Objectives, *o)
```

Comment:
do we catch empty objectives/scenarios elsewhere?

Reply:
This is just in the context of generation, e.g. when you put your QA hat on and give application generation `--goals ""` (which as you also pointed out, I regressed. Again).